### PR TITLE
Automated Publishing to our AWS S3-based CDN

### PR DESCRIPTION
**Pull request by cabutlermit**
### Why these changes are being introduced

We will have some repositories with "static" web content that will need to be published to our S3-based CDN in our AWS Organization. For now, we just have one ([future-of-libraries-static](https://github.com/mitlibraries/future-of-libraries-static)), but there will be more. The publishing of content from the repo to S3 can be handled via GitHub Actions, and it makes the most sense to build this automation as a shared workflow that can be invoked from other repos.

This workflow was tested. You can see the results in the [Actions tab](https://github.com/MITLibraries/future-of-libraries-static/actions/runs/3708414824) on the ([future-of-libraries-static](https://github.com/mitlibraries/future-of-libraries-static)) repository.

### How this addresses that need

* Create the `cdn-shared-publish.yml` workflow to sync repo files with folders in the S3 bucket

### Side effects of this change

No side effects. No dependency changes.

### Relevant ticket(s)

* https://mitlibraries.atlassian.net/browse/LM-93

### CODE REVIEW

**Review by tim-rix — APPROVED** [review]
Everything looks OK to me

File: .github/workflows/cdn-shared-publish.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: PROD Configure AWS credentials
+        # Only run this step if the environment is "prod"
+        if: ${{ inputs.ENVIRONMENT == 'prod' }}
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCT_PROD }}:role/${{ inputs.GHA_ROLE }}
+          aws-region: ${{ inputs.AWS_REGION }}
+
+      - name: Sync S3 content
+        run: aws s3 sync . ${{ inputs.S3URI }} --delete --exclude ".github/*" --exclude ".git/*" --exclude ".gitignore"
```
Comment (JPrevost) [inline .github/workflows/cdn-shared-publish.yml:61]: I'm curious if we could allow overriding `exclude` via config in the calling workflow to provide more flexibility.

Everything currently here I suspect we'd want excluded for all static sites, but I can imagine some might have additional directories or files to exclude such as `docs` or even just the standard `README.MD`. This idea isn't blocking and we could update this when we have a clearer specific use case in the future.

**Review by JPrevost — APPROVED** [review]
I added a comment on whether we might want to allow more flexibility for `excludes` but that feels like something we can adjust as we learn more about repos using this workflow

File: .github/workflows/cdn-shared-publish.yml (hunk 1 above)
Comment (cabutlermit) [reply .github/workflows/cdn-shared-publish.yml:61]: > I'm curious if we could allow overriding `exclude` via config in the calling workflow to provide more flexibility.
> 
> Everything currently here I suspect we'd want excluded for all static sites, but I can imagine some might have additional directories or files to exclude such as `docs` or even just the standard `README.MD`. This idea isn't blocking and we could update this when we have a clearer specific use case in the future.

I was thinking some of the same thoughts (particularly excluding a README.md file! I think we are fine for now, but we will definitely need to revisit this in the future.

**Pull request merged**